tachyon/base/containers/container_util.h

```diff
@@ -97,6 +98,42 @@ std::vector<ReturnType> CreateVectorParallel(size_t size,
   return ret;
 }
 
+template <typename Generator,
+          typename FunctorTraits = internal::MakeFunctorTraits<Generator>,
+          typename ReturnType = typename FunctorTraits::ReturnType,
+          typename RunType = typename FunctorTraits::RunType,
+          typename ArgList = internal::ExtractArgs<RunType>,
+          size_t ArgNum = internal::GetSize<ArgList>,
+          std::enable_if_t<ArgNum == 0>* = nullptr>
+std::pmr::vector<ReturnType> CreatePmrVector(size_t size,
+                                             Generator&& generator) {
+  std::pmr::vector<ReturnType> ret;
+  ret.reserve(size);
+  std::generate_n(std::back_inserter(ret), size,
+                  std::forward<Generator>(generator));
```

Comment:
Is this code inefficient?

```suggestion
  std::pmr::vector<ReturnType> ret(size);
  std::generate_n(ret.begin(), size, std::forward<Generator>(generator));
```

Reply:
This will actually construct `size` number of elements which is equivalent of `resize()`. But here we just `reserve()`.